src/v/storage/segment_appender_chunk.h

```diff
@@ -24,11 +25,10 @@
 namespace storage {
 class segment_appender_chunk {
 public:
-    static constexpr const size_t chunk_size = 16_KiB;
-
-    explicit segment_appender_chunk(size_t alignment)
+    explicit segment_appender_chunk(size_t alignment, size_t size)
       : _alignment(alignment)
-      , _buf(ss::allocate_aligned_buffer<char>(chunk_size, alignment)) {
+      , _chunk_size(size)
+      , _buf(ss::allocate_aligned_buffer<char>(_chunk_size, alignment)) {
```

Comment:
Small comment: there is some potential for confusion here as the order in the constructor is different than the order in the seastar function

Reply:
I guess for any reasonable values the alignment check would fail. but I agree its a little confusing

Reply:
I've swapped these now.